org-brain.el

```diff
@@ -93,39 +165,54 @@ If RELATIVE is t, then return relative paths and remove org extension."
 
 (defun org-brain-parents (entry)
   "Get list of org-brain entries which links to ENTRY."
-  (-non-nil
-   (mapcar
-    (lambda (brainfile)
-      (let ((brainfile-entry (org-brain-path-entry-name brainfile)))
-        (unless (string-equal brainfile-entry entry)
-          (org-element-map
-              (with-temp-buffer
-                (insert-file-contents brainfile)
-                (org-element-parse-buffer))
-              'link
-            (lambda (link)
-              (when (and (string-equal (org-element-property :type link) "brain")
-                         (string-equal (car (split-string (org-element-property :path link) "::"))
-                                       entry))
-                brainfile-entry))
-            nil t))))
-    (org-brain-files))))
+  (if (and org-brain-parents-cache
+           (assoc entry org-brain-parents-cache))
+      (cdr (assoc entry org-brain-parents-cache))
+    (progn
```

Comment:
I think the `progn` isn't needed, since the `else` clause is a `progn` in itself (so to speak).